crypto/bio/bio_sock.c

```diff
@@ -451,5 +452,22 @@ int BIO_socket_wait(int fd, int for_read, time_t max_time)
     tv.tv_sec = (long)(max_time - now); /* might overflow */
     return select(fd + 1, for_read ? &confds : NULL,
                   for_read ? NULL : &confds, NULL, &tv);
+#else
+    struct pollfd confds;
+    time_t now;
+
+    if (fd < 0)
+        return -1;
+    if (max_time == 0)
+        return 1;
+
+    now = time(NULL);
```

Comment:
This should probably use ossl_time_now instead of time directly.  The select code likely should too, but that quite old code.

Reply:
You might be right, but IMHO this should be addressed in another (separate) commit as it would require more changes if time_t is replaced by OSSL_TIME.

Reply:
I agree with both - the interface to `BIO_socket_wait()` uses `time_t`, so any changes to use ossl_time should be done separately